test/core/security/grpc_crl_provider_test.cc

```diff
@@ -28,23 +28,37 @@
 #include <grpc/grpc_audit_logging.h>
 #include <grpc/grpc_crl_provider.h>
 
+#include "src/core/lib/security/credentials/tls/grpc_tls_crl_provider.h"
 #include "test/core/util/test_config.h"
 #include "test/core/util/tls_utils.h"
 
 #define CRL_PATH "test/core/tsi/test_creds/crl_data/crls/ab06acdd.r0"
+#define CRL_ISSUER "/C=AU/ST=Some-State/O=Internet Widgits Pty Ltd/CN=testca"
 
 namespace grpc_core {
 namespace testing {
 
 using experimental::CertificateInfo;
 using experimental::Crl;
+using experimental::CrlImpl;
 using experimental::CrlProvider;
 
 TEST(CrlProviderTest, CanParseCrl) {
   std::string crl_string = GetFileContents(CRL_PATH);
   absl::StatusOr<std::shared_ptr<Crl>> result = Crl::Parse(crl_string);
   ASSERT_TRUE(result.ok());
   ASSERT_NE(*result, nullptr);
+  std::shared_ptr<CrlImpl> crl = std::static_pointer_cast<CrlImpl>(*result);
+  const X509_CRL* x509_crl = &crl->crl();
+  X509_NAME* issuer = X509_CRL_get_issuer(x509_crl);
+  const char* buf = X509_NAME_oneline(issuer, nullptr, 0);
+  ASSERT_EQ(std::string(buf), CRL_ISSUER);
+}
+
+TEST(CrlProviderTest, InvalidFile) {
+  std::string crl_string = "INVALID CRL FILE";
+  absl::StatusOr<std::shared_ptr<Crl>> result = Crl::Parse(crl_string);
+  ASSERT_FALSE(result.ok());
```

Comment:
Please check the status code and message instead of just that it's not okay.

Reply:
Added a check for the type and message

In grpc do we make it a habit of checking error messages or just the code? I know error string testing can be brittle

Reply:
In terms of our public API, we don't guarantee that these messages won't change between versions, so users can't count on any specific string here.  However, we do want to make sure that when we return an error to the user, the error message does contain a useful description and any pertinent info that they need for debugging, so we do want to cover this in our tests.  That means that if we change the code to produce different error messages, we need to update tests, but in practice that hasn't been a major problem.